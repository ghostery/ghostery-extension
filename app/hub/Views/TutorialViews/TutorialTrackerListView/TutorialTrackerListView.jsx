/**
 * Tutorial Tracker List View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 *
 * ToDo: Update this file.
 */

import React from 'react';

/**
 * @class Implement the Tutorial Tracker List View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
const TutorialTrackerListView = () => <div>Ghostery Hub - Tutorial Tracker List View</div>;

export default TutorialTrackerListView;
