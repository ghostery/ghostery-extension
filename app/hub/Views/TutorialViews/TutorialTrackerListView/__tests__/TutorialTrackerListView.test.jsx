/**
 * Tutorial Tracker List View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TutorialTrackerListView from '../TutorialTrackerListView';

describe('app/hub/Views/TutorialViews/TutorialTrackerListView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('tutorial tracker list view is rendered correctly', () => {
			const component = renderer.create(<TutorialTrackerListView />).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const component = shallow(<TutorialTrackerListView />);
			expect(component.find('.TutorialTrackerListView').length).toBe(1);
			expect(component.find('.TutorialTrackerListView__image').length).toBe(1);
		});
	});
});
