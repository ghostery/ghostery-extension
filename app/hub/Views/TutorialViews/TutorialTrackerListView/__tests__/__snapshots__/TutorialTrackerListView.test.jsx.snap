// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/TutorialViews/TutorialTrackerListView component Snapshot tests with react-test-renderer tutorial tracker list view is rendered correctly 1`] = `
<div
  className="TutorialTrackerListView TutorialView--mediumFlexColumn row align-center flex-container"
>
  <div
    className="columns small-12 medium-9"
  >
    <img
      alt="hub_tutorial_trackerlist_title"
      className="TutorialTrackerListView__image shrink"
      src="/app/images/hub/tutorial/tracker-list.png"
    />
  </div>
  <div
    className="TutorialView--paddingTopSmall columns small-10 medium-6 large-3"
  >
    <div
      className="TutorialView__title"
    >
      hub_tutorial_trackerlist_title
    </div>
    <div
      className="TutorialView__tagline"
    >
      hub_tutorial_trackerlist_tagline
    </div>
  </div>
</div>
`;
